api/grpc/mpi/v1/command.proto

```diff
@@ -0,0 +1,387 @@
+        INSTANCE_TYPE_UNIT = 4;
+    }
+    // the types of instances possible
+    InstanceType instance_type = 2;
+    // the version of the instance
+    string version = 3;
+}
+
+// Health report of a set of instances
+message UpdateDataPlaneHealthRequest {
+    // Meta-information associated with a message
+    mpi.v1.MessageMeta message_meta = 1;
+    // Health report of a set of instances
+    repeated InstanceHealth instance_healths = 2;
+}
+
+// Response to a UpdateDataPlaneHealthRequest - intentionally empty
+message UpdateDataPlaneHealthResponse {}
+
+// Reports the status of an associated command. This may be in response to a ManagementPlaneRequest
+message DataPlaneResponse {
+    // Meta-information associated with a message
+    mpi.v1.MessageMeta message_meta = 1;
+    // The command response with the associated request
+    mpi.v1.CommandResponse command_response = 2;
+}
+
+// A Management Plane request for information, triggers an associated rpc on the DataPlane
+message ManagementPlaneRequest {
+    // Meta-information associated with a message
+    mpi.v1.MessageMeta message_meta = 1;
+    oneof request {
+        // triggers a DataPlaneStatus rpc
+        StatusRequest status_request = 2;              
+        // triggers a DataPlaneHealth rpc
+        HealthRequest health_request = 3;               
+        // triggers a rpc GetFile(FileRequest) for overview list, 
+        // if overview is missing, triggers a rpc GetOverview(ConfigVersion) first
+        ConfigApplyRequest config_apply_request = 5;
+        // triggers a series of rpc UpdateFile(File) for that instances
+        ConfigUploadRequest config_upload_request = 6;
+        // triggers a DataPlaneResponse with a command_response for a particular action
+        APIActionRequest action_request = 7;
+        // triggers a DataPlaneResponse with a command_response for a particular correlation_id
+        CommandStatusRequest command_status_request = 8; 
+    }
+}
+
+// Additional information associated with a StatusRequest
+message StatusRequest {}
+
+// Additional information associated with a HealthRequest
+message HealthRequest {}
+
+// Additional information associated with a ConfigApplyRequest
+message ConfigApplyRequest {
+    // the config version
+    mpi.v1.ConfigVersion config_version = 1;
+    // an optional set of files related to the request
+    optional mpi.v1.FileOverview overview = 2;
+}
+
+message ConfigSyncRequest {
+    // the config version
+    mpi.v1.ConfigVersion config_version = 1;
+    // an optional set of files related to the request
+    optional mpi.v1.FileOverview overview = 2;
+}
+
+// Additional information associated with a ConfigUploadRequest
+message ConfigUploadRequest {
+    // the instance identifier
+    string instance_id = 1;
+    // an optional set of files related to the request
+    optional mpi.v1.FileOverview overview = 2;
+}
+
+// Perform an associated API action on an instance
+message APIActionRequest {
```

Comment:
thinking about NGINXaaS' azure-specific features, I think this "API" feature might be one option; NGINXaaS could run a local web server on the dataplane that the agent hits via these `APIActionRequest`s to do our weird azure stuff without needing to pollute various `enum`s here.

Are there intentions to expand this to allow specifying URLs and request bodies for the API action, and for the API response to make it back up to the management plane as part of a `CommandResponse`?

Or is the scope of `APIActionRequest` intended to be limited to an allowlist of N+ / Unit requests?